src/_gettsim/policy_environment.py

```diff
@@ -2,193 +2,138 @@
-        Set up the policy environment for a particular date.
-
-        Parameters
-        ----------
-        date:
-            The date for which the policy system is set up. An integer is
-            interpreted as the year.
-
-        Returns
-        -------
-        environment:
-            The policy environment for the specified date.
-        """
-        # Check policy date for correct format and convert to datetime.date
-        date = _parse_date(date)
-
-        params = {}
-        for group in INTERNAL_PARAMS_GROUPS:
-            params_one_group = _load_parameter_group_from_yaml(date, group)
-
-            # Align parameters for piecewise polynomial functions
-            params[group] = _parse_piecewise_parameters(params_one_group)
-
-        # Extend dictionary with date-specific values which do not need an own function
-        params = _parse_kinderzuschl_max(date, params)
-        params = _parse_einführungsfaktor_vorsorgeaufw_alter_ab_2005(date, params)
-        params = _parse_vorsorgepauschale_rentenv_anteil(date, params)
-        functions = load_functions_for_date(date)
-
-        # Load aggregation specs
-        aggregate_by_group_specs = load_internal_aggregation_dict("aggregate_by_group")
-        aggregate_by_p_id_specs = load_internal_aggregation_dict("aggregate_by_p_id")
-
-        return PolicyEnvironment(
-            functions, params, aggregate_by_group_specs, aggregate_by_p_id_specs
-        )
-
     def __init__(
         self,
-        functions: list[PolicyFunction | Callable],
+        policy_functions_tree: NestedFunctionDict,
         params: dict[str, Any] | None = None,
-        aggregate_by_group_specs: dict[str, dict[str, str]] | None = None,
-        aggregate_by_p_id_specs: dict[str, dict[str, str]] | None = None,
+        aggregations_tree: NestedAggregationDict | None = None,
     ):
-        self._functions = {}
-        for function in functions:
-            f = (
-                function
-                if isinstance(function, PolicyFunction)
-                else PolicyFunction(function)
-            )
-            self._functions[f.name_in_dag] = f
+        # Check functions tree and convert functions to PolicyFunction if necessary
+        assert_valid_pytree(
+            policy_functions_tree,
+            lambda leaf: isinstance(leaf, PolicyFunction),
+            "policy_functions_tree",
+        )
+        self._policy_functions_tree = optree.tree_map(
+            func=_convert_all_functions_to_policy_functions,
+            tree=policy_functions_tree,
+        )
 
+        # Read in parameters and aggregation specs
         self._params = params if params is not None else {}
-        self._aggregate_by_group_specs = (
-            aggregate_by_group_specs if aggregate_by_group_specs is not None else {}
-        )
-        self._aggregate_by_p_id_specs = (
-            aggregate_by_p_id_specs if aggregate_by_p_id_specs is not None else {}
+        self._aggregations_tree = (
```

Comment:
```suggestion
        self._aggregation_specs_tree = (
```
Just to be sure, same for the public property.